src/core/cgroup.c

```diff
@@ -2421,6 +2426,12 @@ void unit_release_cgroup(Unit *u) {
         }
 }
 
+static void unit_prune_cgroup_bpf(Unit *u) {
+#if HAVE_LIBBPF
+        u->bpf_device_control_installed = bpf_program_unref(u->bpf_device_control_installed);
+#endif
```

Comment:
such ifdeffery should be left to the implementation of bpf_program_unref(), not the users